firmware/src/sgx/src/trusted/system.c

```diff
@@ -206,12 +206,19 @@ bool system_init(unsigned char *msg_buffer, size_t msg_buffer_size) {
     }
 
     nvmem_init();
-    nvmem_register_block("bcstate",
-                         &N_bc_state_var,
-                         sizeof(N_bc_state_var));
-    nvmem_register_block("bcstate_updating",
-                         &N_bc_state_updating_backup_var,
-                         sizeof(N_bc_state_updating_backup_var));
+    if (!nvmem_register_block("bcstate",
+                              &N_bc_state_var,
+                              sizeof(N_bc_state_var))) {
```

Comment:
Is there no compilation error with the type change to `uint8_t`?

Reply:
Nope, no errors or warnings. I don't think `sizeof` is typed is it?

Reply:
True, `sizeof` is not typed.